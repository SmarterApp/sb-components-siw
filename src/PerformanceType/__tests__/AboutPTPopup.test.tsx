import * as React from "react";
import { shallow } from "enzyme";
import { AboutPTPopupModal } from "../AboutPTPopup";

describe("AboutPTPopupModal", () => {
  const wrapperUnExpanded = shallow(
    <AboutPTPopupModal
      subject="test"
      description="lorem ipsum"
      isPerformance={false}
      showModal={false}
    />,
    { lifecycleExperimental: true }
  );

  const wrapperExpanded = shallow(
    <AboutPTPopupModal
      subject="test"
      description="lorem ipsum"
      isPerformance={false}
      showModal={true}
    />,
    { lifecycleExperimental: true }
  );

  it("non-expanded AboutPT", () => {
    wrapperUnExpanded.findWhere(node => node.type() === AboutPTPopupModal);
    expect(wrapperUnExpanded).toMatchSnapshot();
  });

  it("expanded AboutPT", () => {
    wrapperExpanded.findWhere(node => node.type() === AboutPTPopupModal);
    expect(wrapperExpanded).toMatchSnapshot();
  });

  it("close button header AboutPT", () => {
    wrapperExpanded
      .findWhere(node => node.type() === "button")
      .at(0)
      .simulate("click");
    expect(wrapperExpanded).toMatchSnapshot();
  });

  it("close button footer AboutPT", () => {
    wrapperExpanded
      .findWhere(node => node.type() === "button")
      .at(1)
      .simulate("click");
    expect(wrapperExpanded).toMatchSnapshot();
  });
});
