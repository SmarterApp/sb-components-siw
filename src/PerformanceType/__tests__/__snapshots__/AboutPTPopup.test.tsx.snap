// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPTPopupModal close button footer AboutPT 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="react-modal-content about-pt-modal"
    closeTimeoutMS={0}
    contentLabel=""
    isOpen={false}
    onRequestClose={[Function]}
    overlayClassName="react-modal-overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      aria-describedby="About Performance Tasks"
      aria-hidden="true"
      aria-labelledby="About Performance Tasks"
      className="modal-wrapper"
    >
      <div
        className="modal-header"
      >
        <h4
          className="modal-title"
        />
        <button
          aria-label="Close modal"
          className="close"
          onClick={[Function]}
        >
          <span
            aria-hidden="true"
            className="fa fa-times"
          />
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p
          aria-labelledby="description"
          dangerouslySetInnerHTML={
            Object {
              "__html": "lorem ipsum",
            }
          }
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          aria-label="Close modal"
          className="btn btn-primary"
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </Modal>
</div>
`;

exports[`AboutPTPopupModal close button header AboutPT 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="react-modal-content about-pt-modal"
    closeTimeoutMS={0}
    contentLabel=""
    isOpen={false}
    onRequestClose={[Function]}
    overlayClassName="react-modal-overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      aria-describedby="About Performance Tasks"
      aria-hidden="true"
      aria-labelledby="About Performance Tasks"
      className="modal-wrapper"
    >
      <div
        className="modal-header"
      >
        <h4
          className="modal-title"
        />
        <button
          aria-label="Close modal"
          className="close"
          onClick={[Function]}
        >
          <span
            aria-hidden="true"
            className="fa fa-times"
          />
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p
          aria-labelledby="description"
          dangerouslySetInnerHTML={
            Object {
              "__html": "lorem ipsum",
            }
          }
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          aria-label="Close modal"
          className="btn btn-primary"
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </Modal>
</div>
`;

exports[`AboutPTPopupModal expanded AboutPT 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="react-modal-content about-pt-modal"
    closeTimeoutMS={0}
    contentLabel=""
    isOpen={true}
    onRequestClose={[Function]}
    overlayClassName="react-modal-overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      aria-describedby="About Performance Tasks"
      aria-hidden="true"
      aria-labelledby="About Performance Tasks"
      className="modal-wrapper"
    >
      <div
        className="modal-header"
      >
        <h4
          className="modal-title"
        />
        <button
          aria-label="Close modal"
          className="close"
          onClick={[Function]}
        >
          <span
            aria-hidden="true"
            className="fa fa-times"
          />
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p
          aria-labelledby="description"
          dangerouslySetInnerHTML={
            Object {
              "__html": "lorem ipsum",
            }
          }
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          aria-label="Close modal"
          className="btn btn-primary"
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </Modal>
</div>
`;

exports[`AboutPTPopupModal non-expanded AboutPT 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="react-modal-content about-pt-modal"
    closeTimeoutMS={0}
    contentLabel=""
    isOpen={false}
    onRequestClose={[Function]}
    overlayClassName="react-modal-overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      aria-describedby="About Performance Tasks"
      aria-hidden="true"
      aria-labelledby="About Performance Tasks"
      className="modal-wrapper"
    >
      <div
        className="modal-header"
      >
        <h4
          className="modal-title"
        />
        <button
          aria-label="Close modal"
          className="close"
          onClick={[Function]}
        >
          <span
            aria-hidden="true"
            className="fa fa-times"
          />
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p
          aria-labelledby="description"
          dangerouslySetInnerHTML={
            Object {
              "__html": "lorem ipsum",
            }
          }
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          aria-label="Close modal"
          className="btn btn-primary"
          onClick={[Function]}
        >
          Close
        </button>
      </div>
    </div>
  </Modal>
</div>
`;
